src/main/java/com/example/x3/MovieManagementApp/services/impl/MovieCommentServiceImpl.java

```diff
@@ -0,0 +1,92 @@
+package com.example.x3.MovieManagementApp.services.impl;
+
+import com.example.x3.MovieManagementApp.dtos.MovieCommentsDto.MovieCommentAddDto;
+import com.example.x3.MovieManagementApp.dtos.MovieCommentsDto.MovieCommentDto;
+import com.example.x3.MovieManagementApp.entities.MovieComments;
+import com.example.x3.MovieManagementApp.entities.Movies;
+import com.example.x3.MovieManagementApp.repositories.MovieCommentRepository;
+import com.example.x3.MovieManagementApp.repositories.MovieRepository;
+import com.example.x3.MovieManagementApp.services.MovieCommentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class MovieCommentServiceImpl implements MovieCommentService {
+
+    @Autowired
+    MovieCommentRepository movieCommentRepository;
+
+    @Autowired
+    MovieRepository movieRepository;
+
+    @Override
+    public List<MovieComments> findAll() {
+        return movieCommentRepository.findAll();
+    }
+
+    @Override
+    public MovieComments findById(Long id) {
+        Optional<MovieComments> tempMovieComment = movieCommentRepository.findById(id);
+
+        return tempMovieComment.orElse(null);
+    }
+
+    @Override
+    public List<MovieComments> findAllCommentsByMovieId(Long movieId) {
+        Optional<Movies> tempMovie = movieRepository.findById(movieId);
+
+        if (tempMovie.isPresent()) {
+            return movieCommentRepository.findAllByMovieId(movieId);
+        }
+        else {
+            return Collections.emptyList();
+        }
+    }
+
+    @Override
+    public String saveComment(MovieCommentAddDto movieCommentAddDto) {
+        List<MovieComments> commentsByUser = movieCommentAddDto.getUser().getComments();
+        for (MovieComments x : commentsByUser) {
+            if ((x.getMovie().getId()) == movieCommentAddDto.getMovie().getId()) {
+                return "Error. This user has already made comment for this movie. Choose another movie";
+            }
+        }
+
+        MovieComments newMovieComment = new MovieComments();
+        newMovieComment.setComment(movieCommentAddDto.getComment());
+        movieCommentRepository.save(newMovieComment);
+
+        return "Comment is saved";
+    }
```

Comment:
The movie comments section is for people to comment back and forth about the movie, so they should be able to have any number of comments on the movie.  This functionality (limiting it to one per user) would be for the ratings entity.  For ratings we only want the user to have one review (rating) per movie.